src/WPGraphQL.php

```diff
@@ -605,25 +605,7 @@ public static function get_allowed_post_types( $output = 'names', $args = [] ) {
 				'objects'
 			);
 
-			$post_type_names = [];
-
-			foreach ( $post_type_objects as $post_type_object ) {
-				/**
-				 * Validate that the post_types have a graphql_single_name and graphql_plural_name
-				 */
-				if ( empty( $post_type_object->graphql_single_name ) || empty( $post_type_object->graphql_plural_name ) ) {
-					throw new UserError(
-						sprintf(
-							/* translators: %s will replaced with the registered type */
-							__( 'The %s post_type isn\'t configured properly to show in GraphQL. It needs a "graphql_single_name" and a "graphql_plural_name"', 'wp-graphql' ),
-							$post_type_object->name
-						)
-					);
-				}
-
-				// Save allowed post type names so they can be filtered by the user.
-				$post_type_names[] = $post_type_object->name;
-			}
+			$post_type_names = wp_list_pluck( $post_type_objects, 'name' );
```

Comment:
well, it's still just post types set to `show_in_graphql`, because we call `get_post_types( [ 'show_in_graphql' => true ]);` above. 

So, we initially start with a list of post types set to `show_in_graphql` (even if they mistakenly forget to set graphql_single_name / graphql_plural_name)

Then, we apply this filter over that already reduced list of post types.

After the filter is applied we validate that the post type has at least the graphql_single_name set (and set graphql_plural_name fallback if needed), then also make sure the post type is allowed by comparing to the post-filtered `$allowed_post_types` list. 

I believe it's the same intended functionality as before, but slightly cleaner and (in my opinion) a better user experience for gracefully failing when a post type / taxonomy is registered incorrectly

Reply:
💯 just wanted to leave a note in case the change in behavior requires a local code change.